src/NTDll/NTDll.cs

```diff
@@ -103,6 +103,20 @@ public static partial class NTDll
             int ProcessInformationLength,
             out int ReturnLength);
 
+        /// <summary>
+        /// The RtlGetVersion routine returns version information about the currently running operating system.
+        /// </summary>
+        /// <param name="versionInformation">
+        /// A RTL_OSVERSIONINFOEXW structure that contains the version information about the currently running operating system.
+        /// </param>
+        /// <returns>
+        /// RtlGetVersion returns STATUS_SUCCESS.
+        /// </returns>
+        /// <seealso href="https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-rtlgetversion"/>
+        [DllImport(nameof(NTDll), SetLastError = true)]
+        public static unsafe extern NTSTATUS RtlGetVersion(
+            [Friendly(FriendlyFlags.Bidirectional)] Kernel32.OSVERSIONINFOEX* versionInformation);
```

Comment:
There should be an overload for `OSVERSIONINFO*`. 

I suspect that would cause problems when a `Friendly `overload is generated with the signature `static PInvoke.NTDll.RtlGetVersion(System.IntPtr versionInformation) -> PInvoke.NTSTATUS` twice. 

If you have to suppress `Friendly` overload generation (can it be done?), then suppressing it for the `OSVERSIONINFO*` variant would be the way to go IMHO.

Reply:
Yes, I initially added the `OSVERSIONINFO*` overload, but then the `IntPtr` overloads gets generated twice.

Reply:
BTW the official [documentation](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-rtlgetversion) defines it like this: 

`C

NTSYSAPI NTSTATUS RtlGetVersion(
  PRTL_OSVERSIONINFOW lpVersionInformation
);
`

It doesn't look like there is a version for `PRTL_OSVERSIONINFOEXW` though.

Reply:
Well, [from that same document](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-rtlgetversion#parameters):

> `lpVersionInformation`
> Pointer to either a `RTL_OSVERSIONINFOW` structure or a `RTL_OSVERSIONINFOEXW` structure that contains the version information about the currently running operating system. *A caller specifies which input structure is used by setting the `dwOSVersionInfoSize` member of the structure to the size in bytes of the structure that is used.*

Reply:
Apologies for being unclear. What I meant to write is this - 

The official documentation shows the function signature based on `PRTL_OSVERSIONINFOW` instead of `PRTL_OSVERSIONINFOEXW` (also the headers show this, and debugging into the functions would show stack-frames that match this expectation as well).  I was trying to suggest that it would be more accurate for the `DllImport` declaration to reflect this fact. Even though a `DllImport `declaration with a signature `NTSTATUS RtlGetVersion(PRTL_OSVERSIONINFOEXW)` is likely to **work** just fine because of how the  `dwOSVersionInfoSize` parameter is interpreted, and the fact that `RTL_OSVERSIONINFO` and `RTL_OSVERSIONINfOEX` are layout-compatible with one-another, in the native frame it would first get cast into `RTL_OSVERSIONINFO*`, and then its `dwOSVersionInfoSize` would be read following by reinterpretation again as `RTL_OSVERSIONINFOEXW*`

Reply:
Thanks, @vatsan-madhavan . I think we're on the same page. I updated the P/Invoke declaration to use `OSVERSIONINFO` and added a helper method in `NTDllExtensions` which uses `OSVERSIONINFOEX`, as you suggested.